test/common/operation.go

```diff
@@ -0,0 +1,244 @@
+		return err
+	}
+	return client.sendFakeEventToAddressable(senderName, broker.Status.Address, event)
+}
+
+// sendFakeEventToAddressable will create a sender pod, which will send the given event to the given url.
+func (client *Client) sendFakeEventToAddressable(
+	senderName string,
+	addr duckv1alpha1.Addressable,
+	event *base.CloudEvent,
+) error {
+	namespace := client.Namespace
+	url := fmt.Sprintf("http://%s", addr.Hostname)
+	client.Logf("Sending fake CloudEvent")
+	pod := base.EventSenderPod(senderName, url, event)
+	if err := client.CreatePod(pod); err != nil {
+		return err
+	}
+	if err := pkgTest.WaitForPodRunning(client.Kube, senderName, namespace); err != nil {
+		return err
+	}
+	return nil
+}
+
+// WaitForBrokerReady waits until the broker is Ready.
+func (client *Client) WaitForBrokerReady(name string) error {
+	namespace := client.Namespace
+	brokerMeta := base.MetaObj{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Broker",
+			APIVersion: base.EventingAPIVersion,
+		},
+	}
+	if err := base.WaitForResourceReady(client.Dynamic, brokerMeta); err != nil {
+		return err
+	}
+	return nil
+}
+
+// WaitForBrokersReady waits until all brokers in the namespace are Ready.
+func (client *Client) WaitForBrokersReady() error {
+	namespace := client.Namespace
+	brokers, err := client.Eventing.EventingV1alpha1().Brokers(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, broker := range brokers.Items {
+		if err := client.WaitForBrokerReady(broker.Name); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// WaitForTriggerReady waits until the trigger is Ready.
+func (client *Client) WaitForTriggerReady(name string) error {
+	namespace := client.Namespace
+	triggerMeta := base.MetaObj{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Trigger",
+			APIVersion: base.EventingAPIVersion,
+		},
+	}
+	if err := base.WaitForResourceReady(client.Dynamic, triggerMeta); err != nil {
+		return err
+	}
+	return nil
+}
+
+// WaitForTriggersReady waits until all triggers in the namespace are Ready.
+func (client *Client) WaitForTriggersReady() error {
+	namespace := client.Namespace
+	triggers, err := client.Eventing.EventingV1alpha1().Triggers(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, trigger := range triggers.Items {
+		if err := client.WaitForTriggerReady(trigger.Name); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// WaitForChannelReady waits until the channel is Ready.
+func (client *Client) WaitForChannelReady(name string) error {
+	namespace := client.Namespace
+	channelMeta := base.MetaObj{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Channel",
+			APIVersion: base.EventingAPIVersion,
+		},
+	}
```

Comment:
Again, not required, but you could define an `objectMeta(client.Namespace, name, "Channel")` function to get this object.

Reply:
This has already been done in one commit before your comment :-)